### ISSUE TITLE
Computer Modern Roman should be the default serif when using TeX backend

### ISSUE THREAD

**Issue by srvanrell** [matplotlib/matplotlib#5860]
I come from this question http://stackoverflow.com/questions/16287921/python-matplotlib-change-axis-labels-legend-from-bold-to-regular-weight

There, Alexander Vogt explains that the default serif font is not Computer Modern Roman (as it is in any LaTeX document). Instead, the default is New Century School Book, and Alexander claims that it is set by texmanager.py (https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/texmanager.py). I think that maybe it wasn't the original intention when texmanager.py was written, because of the line:

    serif = ('cmr', '')

and what is said in the old issue: https://github.com/matplotlib/matplotlib/issues/882

Don't you agree with Computer Modern Roman as the default serif font?.
Meanwhile, the following trick is a workaround:

    font = {'family':'serif', 'serif': ['computer modern roman']}
    plt.rc('font',**font)

**Comment by tacaswell** [comment]
@zblz @mdboom

**Comment by srvanrell** [comment]
Perhaps the ones should be changed to zeros is in this line https://github.com/matplotlib/matplotlib/blob/v2.x/lib/matplotlib/texmanager.py#L215.

I would gladly contribute to it but I don't know how to test matplotlib source (I cloned the repository, and tried to follow http://matplotlib.org/devel/testing.html but nose is not properly working)

**Comment by srvanrell** [comment]
My previous comment is wrong. The problem is in the [default values](https://github.com/matplotlib/matplotlib/blob/v2.x/lib/matplotlib/rcsetup.py#L908) of font.serif, font.sans-serif, font.monospace (I don't know if font.cursive and font.fantasy need to be changed too). Computer Modern families are not in the lists, so the first known family is loaded instead. The following may be the solution, adding `Computer Modern *` in third position (should be second?) 

Starting at [line 908 in rcsteup.py](https://github.com/matplotlib/matplotlib/blob/v2.x/lib/matplotlib/rcsetup.py#L908): 

    'font.serif':      [['Bitstream Vera Serif', 'DejaVu Serif', 
                         'Computer Modern Roman',
                         'New Century Schoolbook', 'Century Schoolbook L',
                         'Utopia', 'ITC Bookman', 'Bookman',
                         'Nimbus Roman No9 L', 'Times New Roman',
                         'Times', 'Palatino', 'Charter', 'serif'],
                        validate_stringlist],
    'font.sans-serif': [['Bitstream Vera Sans', 'DejaVu Sans', 
                         'Computer Modern Sans Serif',
                         'Lucida Grande', 'Verdana', 'Geneva', 'Lucid',
                         'Arial', 'Helvetica', 'Avant Garde', 'sans-serif'],
                        validate_stringlist],
    'font.cursive':    [['Apple Chancery', 'Textile', 'Zapf Chancery', 
                         'Sand', 'Script MT', 'Felipa', 'cursive'],
                        validate_stringlist],
    'font.fantasy':    [['Comic Sans MS', 'Chicago', 'Charcoal', 'Impact',
                         'Western', 'Humor Sans', 'fantasy'],
                        validate_stringlist],
    'font.monospace':  [['Bitstream Vera Sans Mono', 'DejaVu Sans Mono', 
                         'Computer Modern Typewriter',
                         'Andale Mono', 'Nimbus Mono L', 'Courier New',
                         'Courier', 'Fixed', 'Terminal', 'monospace'],
                        validate_stringlist],

The other thing that I don't know how to handle but it still needed is that when `text.usetex` is set to `True`, then `font.family` should be changed to `serif` (because it is the default for LaTeX documents)

**Comment by tacaswell** [comment]
@srvanrell Can you put in a PR with that change?

**Comment by srvanrell** [comment]
Do you mean the changes in the rcsetup.py? I can do that. Should the added
font be in second position?
El 30 may. 2016 21:17, "Thomas A Caswell" <notifications@github.com>
escribió:

@srvanrell <https://github.com/srvanrell> Can you put in a PR with that
change?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/matplotlib/matplotlib/issues/5860#issuecomment-222568405>,
or mute the thread
<https://github.com/notifications/unsubscribe/AB1Nz1ERRhIQx71SV0yNQ_i-N54iPCTgks5qG34igaJpZM4HF0jv>
.

**Comment by QuLogic** [comment]
I think third position is fine; Bitstream and DejaVu are basically equivalent.

**Comment by zblz** [comment]
Yes, it should be in third position to retain the DejaVu/Bitstream as default if one of these is not installed.

**Comment by srvanrell** [comment]
I've just put that changes in the PR #6512.

However, there is one thing missing that I don't know where it should be fixed. When `text.usetex` is set to `True`, then `font.family` should be changed to `serif` (because it is the default for LaTeX documents)

**Comment by efiring** [comment]
I don't think we want to try to have font.family change automatically based on text.usetex.  They are independent parameters.  It is up to the user to set each one as desired.

**Comment by srvanrell** [comment]
I'm a TeX user and that surely changes my point of view. Do what you consider is the proper behavior for matplotlib

**Comment by zblz** [comment]
AFAICT, The TeX backend main goal is not to have consistency with the rest of TeX, but to use TeX typesetting in plots that are consistent with the matplotlib defaults. For strict TeX consistency you can use the pgf backend, so that the text will be typeset to be consistent with the TeX document in which you are including the plot.

**Comment by srvanrell** [comment]
I leave a minimal working example below, just in case it is useful:
```
import matplotlib.pyplot as plt
import numpy as np

plt.rc('text',usetex=True)

font = {'family':'serif','size':16}  # desired use
# font = {'family':'serif','size':16, 'serif': ['computer modern roman']}  # what you need to do now 
# font = {'size':16} # desired use
# font = {'size':16, 'sans-serif': ['computer modern sans-serif']} # what you need to do now 

plt.rc('font',**font)

x = np.linspace(0,2*np.pi,100)
y = np.sin(x)

plt.figure(1)
plt.plot(x,y)
plt.title('Sample')
plt.show()
```

**Comment by tacaswell** [comment]
Closed via #6512
